nni/experiment/experiment.py

```diff
@@ -377,6 +378,7 @@ def update_search_space(self, value: dict):
         value: dict
             New search_space.
         """
+        value = json.dumps(value)
```

Comment:
I think it should use json_tricks here. It might contains numpy value or infinity/nan.

Reply:
fix it